common/src/main/java/com/microsoft/identity/common/internal/cache/AccountCredentialCache.java

```diff
@@ -204,38 +170,38 @@ public synchronized Credential getCredential(final String cacheKey) {
             }
         }
 
-        Logger.exiting(TAG, methodName, accounts);
+        Logger.verbose(TAG, "Returning [" + accounts.size() + "] Accounts w/ keys...");
 
         return accounts;
     }
 
     @Override
+    @NonNull
     public synchronized List<Account> getAccounts() {
-        final String methodName = "getAccounts";
-        Logger.entering(TAG, methodName);
-
+        Logger.verbose(TAG, "Loading Accounts...(no arg)");
         final Map<String, Account> allAccounts = getAccountsWithKeys();
         final List<Account> accounts = new ArrayList<>(allAccounts.values());
-
-        Logger.exiting(TAG, methodName, accounts);
-
+        Logger.info(TAG, "Found [" + accounts.size() + "] Accounts...");
         return accounts;
     }
 
     @Override
+    @NonNull
     public List<Account> getAccounts(
```

Comment:
Should we consider a different method name here? Not sure if Java or Android has naming guidance when you're filtering a list....

Reply:
How does `findAccounts()` strike you? Trying to express that there's some notion of criteria...

Reply:
We're really filtering an existing collection here.  Would be great to be able to use lambdaj or the Java 8 builtin support for the same.... anyway.... getAccountsFilteredBy?

Reply:
For API 24 this is provided 'for free' via the [stream api](https://developer.android.com/reference/java/util/stream/package-summary) -- once our minSdk updates passes this point, we can bring it into our own code as well. It's possible Google may provide support for this in the future via a support package, though it hasn't happened yet.

`getAccountsFilteredBy` works for me... I'll refrain from introducing a 3rd party stream dependency for now

Reply:
There's similar code for Credentials as well, I'll make an equivalent change there...